scholarx/archive/functions.js

```diff
@@ -52,13 +58,128 @@ async function getData() {
 }
 
 async function loadData() {
-    const payload = await getData();
-    const mentorProfiles = payload.mentor2021.concat(payload.mentor2020).concat(payload.mentor2019);
-    let mentor_Profiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfiles });
-    $("#mentorProfiles").html(mentor_Profiles);
+    const { data } = await getData();
+    mentor = data.mentors;
+    mentee = data.mentees;
+    mentorProfilesData = mentor[2019].concat(mentor[2020]).concat(mentor[2021]);
+    let mentorProfiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfilesData });
+    $("#mentorProfiles").html(mentorProfiles);
 
-    const menteeProfiles = payload.mentee2021.concat(payload.mentee2020).concat(payload.mentee2019);
-    let mentee_Profiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfiles });
-    $("#menteeProfiles").html(mentee_Profiles);
+    menteeProfilesData = mentee[2019].concat(mentee[2020]).concat(mentee[2021]);
+    let menteeProfiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfilesData });
+    $("#menteeProfiles").html(menteeProfiles);
 }
 loadData();
+function renderProfiles(mentorYear,menteeYear) {
+    let mentorProfiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorYear });
+    let menteeProfiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeYear });
+    $("#mentorProfiles").html(mentorProfiles);
+    $("#menteeProfiles").html(menteeProfiles);
+}
+function renderAllProfiles() {
+    let mentorProfiles = Mustache.render($("#templateMentors").html(), { "mentorProfiles": mentorProfilesData });
+    let menteeProfiles = Mustache.render($("#templateMentees").html(), { "menteeProfiles": menteeProfilesData });
+    $("#mentorProfiles").html(mentorProfiles);
+    $("#menteeProfiles").html(menteeProfiles);
+}
+//function to return true if all industry checkboxes are unchecked.
+function allFalse(){
+    let number = 0;
+    for(let i=1; i<7; i++){
+        if(document.getElementById("industry_"+i).checked == false){
+            number++;
+        }
+    }
+    if(number==6){
+        return true
+    }else{
+        return false
+    }
+}
```

Comment:
We can simplify this function

Hint: We can run a loop through all the industry checkboxes (i=0 to i=7). If any of those are checked, we can break the loop and return false